### ISSUE TITLE
Review of lesson 2—it is a big jump from lesson 1; we need to fill the gap

### ISSUE THREAD

**Issue by labarba** [barbagroup/JITcode-MechE#19]
I've read over the (currently) Lesson 2 draft, and I think that this made a large, uncomfortable leap from Lesson 1. In fact, this material might be appropriate for a much  later lesson. The basic objective was to introduce custom functions, but there's a _lot_ more going on. 
1. it uses magics to run a separate python script (with no explanation);
2. the function definition is introduced with an abstract example where 'turn right' and 'go forward' are messages printed out, and nothing is computed;
3. the lesson imports the custom function 'gorover' with no explanation;
4. the function 'student_func' is then passed as an argument to the 'gorover' function; what does that _mean_ to our novice students working through the lesson?
5. it uses a JSAnimation with no explanation.

Basically, I think that this lesson will be full of mysteries and somewhat frustrating for a novice. As a general rule, my "design guide" for these lessons is that each one should introduce _one_ major new concept, and ideally it should do so while reinforcing or repeating something that was already done in a previous lesson. This gives a sense of confidence to the novice and reduces cognitive load.

In summary, this lesson is nice, although it can be improved, but I see it as a lesson 4 or 5 or even 6 … definitely not 2.
